### ISSUE TITLE
[MacOS] Magnet Graphical Bug

### ISSUE THREAD

**Issue by Vitaliaosu** [ppy/osu#12069]
**Description:**
I get huge freezes when trying to use magnet to resize the window.
**THIS ALSO OCCURS WHEN PLAYING, IT'S NOT ONLY IN THE EDITOR**

**Screenshots or videos showing encountered issue:** 
https://www.youtube.com/watch?v=HRGKa68enVQ

**Version:**
v. 2021.312.0

**Comment by Vitaliaosu** [comment]
**MacOS version:**
![Screenshot 2021-03-18 at 23 18 35](https://user-images.githubusercontent.com/69867797/111704993-457c7180-8840-11eb-8d4b-8b1ffdf84b4c.png)

Note: I'm running a Hackintosh, but that shouldn't be an issue.

**Comment by bdach** [comment]
This is unlike most issues I've ever seen. Going off the state of the game when it's "bugged", looks like it's displaying a half-rendered frame.

If you can attach some logs after a crash, maybe we can get somewhere with this report. Otherwise I'm not really even sure in which direction to proceed.

**Comment by Vitaliaosu** [comment]
I’m gonna try to make it crash, so far it’s just constant freezing that is most noticeable in the editor.

**Comment by peppy** [comment]
We can not support hackintoshes, sorry. This looks 99% a GPU driver issue.

**Comment by Vitaliaosu** [comment]
I want to comment that I have absolutely no problem in other programs that use my GPU to render things. I have a 5700 XT which is fully supported in the latest version of Big Sur.
I've tested more maps and I came across a recent tech map, I scroll through the green lines and I almost instantly freeze.

https://osu.ppy.sh/beatmapsets/1317690#osu/2730082

**Here's a demonstration.**
https://www.youtube.com/watch?v=b9BGOmhZqLY
*It clearly shows that the green ticks causes a performance drop which I'm pretty sure is not caused by the fact that I have a hackintosh.
